content/tutorials/numpy_integration/grass_numpy_integration.qmd

```diff
@@ -0,0 +1,253 @@
+the coordinates of the upper-right (nort-east) corner are number of rows times cell resolution and number of columns times cell resolution.
+
+```{python}
+rows = 200
+cols = 250
+resolution = 10
+tools.g_region(s=0, w=0, n=rows * resolution, e=cols * resolution, res=resolution)
+```
+
+## Creating a fractal surface as a NumPy array
+
+We will create a simple fractal surface with [r.surf.fractal](https://grass.osgeo.org/grass-stable/manuals/r.surf.fractal.html).
+
+{{< fa wand-magic-sparkles >}} The trick is to use the `output` parameter with the value `np.array` to request a NumPy array instead of a native GRASS raster. {{< fa wand-magic-sparkles >}}
+This way GRASS provides the array as the result of the call:
+
+```{python}
+import numpy as np
+
+fractal = tools.r_surf_fractal(output=np.array, seed=6)
+```
+
+::: {.callout-note title="NumPy arrays in GRASS version < 8.5"}
+
+Directly passing NumPy arrays to GRASS tools and receiving them back is a new feature in GRASS v8.5.
+If you work with older versions of GRASS, you can use
+[grass.script.array](https://grass.osgeo.org/grass-stable/manuals/libpython/grass.script.html#script.array.array) and [grass.script.array3d](https://grass.osgeo.org/grass-stable/manuals/libpython/grass.script.html#script.array.array3d):
+
+```{python}
+import grass.script as gs
+import grass.script.array as ga
+
+gs.run_command("r.surf.fractal", seed=6, output="fractal")
+fractal = ga.array("fractal")
+```
+
+:::
+
+Now we can display `fractal` array e.g., using matplotlib library:
+
+```{python}
+import matplotlib.pyplot as plt
+
+plt.figure()
+plt.imshow(fractal, cmap='terrain')
+plt.colorbar(label='Elevation')
+plt.title('Topography from r.surf.fractal')
+plt.xlabel('X-coordinate')
+plt.ylabel('Y-coordinate')
+plt.show()
+```
+
+![Fractal surface generated with r.surf.fractal displayed using matplotlib](fractal_numpy.webp)
+
+We can modify the array:
+
+```{python}
+fractal *= 0.1
+fractal = np.abs(fractal)
+
+plt.figure()
+plt.imshow(fractal, cmap='terrain')
+plt.colorbar(label='Elevation')
+plt.title('Modified topography from r.surf.fractal')
+plt.xlabel('X-coordinate')
+plt.ylabel('Y-coordinate')
+plt.show()
+```
+
+![Modified fractal surface](fractal_numpy_abs.webp)
+
+# From GRASS to Landlab
+
+Now, let's use Landlab's modeling capabilities to burn in an initial drainage network using the
+Landlab's [Fastscape Eroder](https://landlab.readthedocs.io/en/latest/generated/api/landlab.components.stream_power.fastscape_stream_power.html).
+
+Create the `RasterModelGrid` specifying the same dimensions we used for creating the fractal surface.
+Add the terrain elevations as a 1D array (flattened with `ravel`) to the grid's nodes under the standard field name "topographic__elevation".
+
+```{python}
+from landlab import RasterModelGrid
+
+grid = RasterModelGrid((rows, cols), xy_spacing=resolution)
```

Comment:
Would shape work here, but I guess rows, cols is consistent with the use of resolution.

```suggestion
grid = RasterModelGrid(fractal.shape, xy_spacing=resolution)
```